sdk/core/core/inc/az_span.h

```diff
@@ -14,6 +15,13 @@
 
 #include <_az_cfg_prefix.h>
 
+#define AZ_CONTRACT_ARG_VALID_SPAN(arg) \
+  do { \
+    if ((arg).begin == NULL && (arg).size > 0) { \
```

Comment:
Use inline function inside this macro. We shouldn't call `arg` twice, it may have side effects.